docs/changelog/0037-upgrade-to-elasticsearch7.md

```diff
@@ -0,0 +1,336 @@
+5. At this point, all of the Elasticsearch 2 data should be in Elasticsearch 7.
+   - You can confirm this by running the `reindex_from_es2_to_es7` management command with `print_index_size` option which displays number of documents in both the ES clusters. 
+7. Run below command to update Elasticsearch settings on ES7 indices for usage
+  ```
+  ./manage.py initialize_es_indices --set-for-usage
+  ```
+8. Update HQ config to point to Elasticsearch 7.
+  - Update below settings in you env's public.yml in addition to other setting changes made in upgrade-branch in the Setup phase
+  ```
+  ELASTICSEARCH_MAJOR_VERSION: 7
+  ```
+  - Update config
+  ```
+  cchq <env> update-config --branch=<upgrade-branch>
+  ```
+    - Merge the upgrade-branch into master  
+9. Make sure that HQ is able to connect to Elasticsearch 7
+  ```
+  cchq <env> django-manage check_services
+  ```
+10. End downtime
+  ```
+  cchq <env> downtime end
+  ```
+That's it, you are running HQ on Elasticsearch 7!
+
+
+
+### Reindex and Switchover separately
+
+If your form/case index sizes are large, you can follow this method to first reindex without downtime and then switchover with downtime.
+
+#### Reindex
+
+Reindex phase involves reindexing data from Elasticsearch 2 cluster into new Elasticsearch 7 cluster. Form and Case indices can be indexed in background while the HQ application continues to be live. This reindex can be performed based on the `inserted_at` date field upto a day before switchover.  Using this parameter multiple reindex processes can be run concurrently for different date ranges enabling you to reindex very large indices. The smaller offset data inserted newly during and after the reindex phase can be reindexed  during the last phase of switchover with downtime. This enables minimising the overall downtime required to upgrade drastically. 
+
+Use below notes to reindex all of the data upto a day or two before switchover and proceed to the switchover section after that.
+
+ 1. The reindex management command should be run from the release that's setup in step 2 of Setup phase above which is pointing to Elasticsearch 7. 
+   - To activate virtual env for reindex commands, SSH into the web worker from Step 2 of Setup phase and do below
+   ```
+  tmux # optional
+  sudo -iu cchq
+  cd <releases_dir_from_setup_phase>
+  source python_env-3.6/bin/activate
+  grep ELASTICSEARCH localsettings.py # to make sure this points to ES7
+   ```
+ 2. Reindex all of form/case data upto a planned switchover date
+     - Below is an example management command that reindexes forms index from Elasticsearch 2 cluster at `http://es2_cluster:9200` for a specific date range
+    ```
+      # to get help on usage
+    ./manage.py reindex_from_es2_to_es7 --help
+    # an example reindex command
+      ./manage.py reindex_from_es2_to_es7 forms http://es2_cluster:9200 --start_date=YYYY-MM-DD
+    --end_date=YYYY-MM-DD 
+   ``` 
```

Comment:
Incorrect indentation here caused improper formatting. I didn't read much below here because the bad formatting made it too hard to read.
```suggestion
    ```
```